vmss-prototype/vmss-prototype

```diff
@@ -522,6 +511,22 @@ def vmss_build_sig_image(subscription, resource_group, sig_name, image_definitio
     return output
 
 
+def format_into_version(date_time):
+    """
+    Convert a datetime into a version string
+    Note that the format of the string is yyyy.mm.dd
+    and is required to be in that kind of format for
+    the SIG to work correctly (defined by Azure)
+    It does not need to be year.month.day but it must
+    be a digits.digits.digits specifically and ordered
+    via the version ordering constraint.
+    :param date_time: The datetime object to convert
+    :return: The version string as we use within the SIG
+    """
+    # Format for version numbers from timestamps
+    return date_time.strftime('%Y.%m.%d')
```

Comment:
Instead of a new func for this can we simply assign the string value of `'%Y.%m.%d'` to a constant, and then invoke `.strftime(<const>)` each time we want to do this conversion?

Reply:
Oh actually I see that's what we *were* doing. :)

Why do we want to prefer the func invocation pattern over the use the const version string pattern?

Reply:
I just like having it as a function.  Constants are a bit nasty in python.

Plus it makes for smaller code :-)

Reply:
Once should never be afraid of a function.  Passing around data to make a separate function perform is, IMHO, slightly less clean if you can encapsulate that into the function, especially in python, as real constants (as in immutable) is actually rather nasty to do (you have to do backflips to prevent re-assignment/re-definition)

Other languages provide a better solution for constants and maybe it would be fine to not define a new function but rather just provide the constant extra argument you need.  (Or do like C# and extension methods that really are functions but look like they are methods on an object - a really nice construct to make clean looking code)

Reply:
PS - the value was needed in yet another place and that would have required a more global "constant" which then leaves you with less confidence that no one plays with it.